### DISCUSSION TITLE
ArgoRollouts and ArgoCD Integration BlueGreen

### DISCUSSION THREAD

**Discussion by rodnavarro** [argoproj/argo-helm#2206] [General]
Hey guys

This is the flow:

- Rollout is defined in a Helm Chart. So I commit a change to the helm chart to change the livenessProbe (as an example). I also have the rollout configured to scale down after 24 hours. 
- Rollout picks the change and it modify the preview service. 
- I promote it now. Works nicely
- Then I want to rollback to the exact same previous pod /
- I rollback. Rollback completes
- ArgoCD Self Heal gets in the way as the Rollout manifest get's updated back (remember the livenessProbe change gets reverted)

It's a race condition now. 

If I disable the self healing for the argocd application, it works. 
The state is out of sync though and I can see the difference from the live state and the git defined state, which makes sence

Is there a way to make this work without having to disable self healing? - so that argocd doesnt try to overwrite? I'm using argocd 2.7.10 and argorollouts 1.4.0. Latest as of Aug 04, 2023

**Comment by yu-croco** [comment]
Hi @rodnavarro , this is related to Argo CD's feature, so can you please open a discussion at [upstream](https://github.com/argoproj/argo-cd)?
argo-helm handles Helm Chart, so I think upstream's discussion will suite to your question.

**Reply by rodnavarro** [reply]
will do! thanks
